### ISSUE TITLE
Caching Probleme bei GridblockContentSettings

### ISSUE THREAD

**Issue by georgkaser** [iceman-fx/gridblock#65]
Servus,

ich habe immer wieder Probleme mit dem Cache - dachte das wäre behoben, aber jetzt tritt es wieder ganz massiv auf verschiedenen Rechnern, verschiedenen Browsern und Umgebungen auf.

Es geht um GridblockContentSettings: Ich ändere einen Wert, drücke auf speichern und eine Überprüfung ergibt, dass die Selects und Checkboxen den alten Wert anzeigen. Eine erneute Speicherung überschreibt den zuvor korrekt gespeicherten Wert. Bei deaktiviertem Browsercache (z.B. wenn die Developertools geöffnet sind) funktionert es wie erwartet. Hier ein Screencast um das Problem zu veranschaulichen:

https://github.com/iceman-fx/gridblock/assets/6472869/31eb201b-a86b-4877-8a41-c4df1e037684

Woran kann das liegen?? So kann man das Addon im Moment nicht einsetzen...

**Comment by georgkaser** [comment]
Ich hab jetzt mal in das Gridblock Modul die Funktion `opcache_reset()` aufgerufen - das müsste ja den Cache zurücksetzen. Leider kein Erfolg...

**Comment by georgkaser** [comment]
Ich bekomme im Gridblock Addon auch massig Fehlermeldungen - vielleicht liegt es daran? 

```
Warning: Undefined array key "" in redaxo/src/addons/gridblock/plugins/contentsettings/lib/class.GridblockContentSettings.php on line 263
Warning: Trying to access array offset on value of type null in redaxo/src/addons/gridblock/plugins/contentsettings/lib/class.GridblockContentSettings.php on line 264
Warning: Undefined array key "" in redaxo/src/addons/gridblock/plugins/contentsettings/lib/class.GridblockContentSettings.php on line 334
Warning: Undefined array key "" in redaxo/src/addons/gridblock/plugins/contentsettings/lib/class.GridblockContentSettings.php on line 891
Warning: Trying to access array offset on value of type null in redaxo/src/addons/gridblock/plugins/contentsettings/lib/class.GridblockContentSettings.php on line 891

```

**Comment by iceman-fx** [comment]
Ich habe bei mir im Redaxo selbst (nicht GB) sehr oft das gleiche Problem, wenn OPCache an ist.
Daher schalte ich es immer ab, auch wenn die Seite dann etwas langsamer wird.

**Comment by iceman-fx** [comment]
Die Fehlermeldungen sollte sich @novinet-dsteffen mal anschauen.
Dürften aber mit deinem Problem nix zu tun haben, da es nur PHP-Warnings sind.

**Comment by georgkaser** [comment]
Also an dem Servercache kann es nicht liegen. Ich hab alle Servercaches deaktiviert und es sogar auf einem anderen Host installiert. Auf diesem hagelt es nur noch Fehlermeldungen. Ich hab das Plugin jetzt rausgeschmissen und bau mir die Settings mittels MForm selber ein. Man sollte das aber auf jedem Fall debuggen, so kann man das Plugin nicht verwenden. Gridblock selber scheint keine Probleme zu machen - es ist definitiv das Plugin.

**Comment by iceman-fx** [comment]
@novinet-dsteffen push

**Comment by dpf-dd** [comment]
Könnte das evtl. auch mit dem zu tun haben, was ich gerade herausgefunden habe?
https://github.com/iceman-fx/gridblock/issues/68

**Comment by iceman-fx** [comment]
@dpf-dd Ich denke eher nicht.

**Comment by iceman-fx** [comment]
Jetzt habe ich selbst das gleiche Problem auf meinem neuen Firmenlaptop.
Es betrifft definitiv das Caching des Browsers in Verbindung mit den Feldern des CS.

Hat jemand von Euch schon eine Ursache oder besser noch Lösung gefunden?

@novinet-dsteffen Evtl. würde es helfen, wenn alle Formfelder immer unterschiedliche IDs und/oder name="" haben würden.
Also eine eindeutige ID oder die SID mit anhängen und auswerten, damit die Felder von Slice zu Slice immer andere Namen haben.

Testszenario bei mir:
2 Gridblöcke mit unterschiedlicher Befüllung der gleichen CS-Felder (z.B. BG-Color).
Dann Block 1 editieren -> gleich danach Block2 -> hier wird mit der Wert aus Block 1 angezeigt.
Jetzt den Browsercache leeren und Block 2 erneut zum editieren öffnen -> korrekter Wert wird angezeigt.

**Comment by georgkaser** [comment]
Nein.

Am 3. Februar 2025 14:48:21 MEZ schrieb "Falko Müller" ***@***.***>:
>@georgkaser Hast du auf deinem Server auch Plesk als Verwaltung?
>Bei mir tritt dies aktuell nur auf meinem Pleskserver auf. 
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/iceman-fx/gridblock/issues/65#issuecomment-2631055400
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
Webdesign & Webprogrammierung
Georg Kaser

Schillerstraße. 5, 67547 Worms
Mobil: +49 (0)157 84925965
https://www.georgkaser.com

**Comment by iceman-fx** [comment]
@georgkaser Ich habe vermutlich eine Lösung gefunden.
Hast Du es noch im Einsatz, um es bei Dir mal testen zu können?

**Comment by iceman-fx** [comment]
Ich habe es in der aktuellen Github-Version einmal hochgeladen.
Bitte gern mal testen.

**Comment by georgkaser** [comment]
> [@georgkaser](https://github.com/georgkaser) Ich habe vermutlich eine Lösung gefunden. Hast Du es noch im Einsatz, um es bei Dir mal testen zu können?

Nein, nachdem ja ich der einzige war, bei dem der Fehler auftrat und niemand sich mit dem Fehler befassen wollte, konnte ich das Addon leider nicht mehr bei meinen Projekten einsetzen. Mir war das Risiko zu hoch, dass die Kunden auf diesen komischen Fehler stoßen und ich keine Lösung anbieten kann.
